### PR TITLE
add subscriptions command

### PR DESCRIPTION
**Pull request by ewindisch**
Breaks subscriptions off into a separate command

Signed-off-by: Erica Windisch <ewindisch@newrelic.com>

### CODE REVIEW

**Comment by ewindisch** [comment]
Right now, most users will be setting up the integration and the subscription with this tool, but not using it to set the layers. We have users migrating from the earlier tools where these were separated. We also have users that will want to integrate these processes into SAM, etc.

Right now, all of the top-level commands do one thing. While it's more complex to give the users multiple steps, it's also consistent and clear.

I am 100% for having a single unified installation command, but given the current use-cases of each of the commands of this tool, I think that should be a new wizard-like command option rather than being built-in to 'layers'.

File: README.md [hunk 1]
```diff
@@ -132,6 +132,37 @@ newrelic-lambda functions list --filter installed
...
+
+```bash
+newrelic-lambda subscriptions install \
+    --function <name or arn> \
+    --nr-account-id <new relic account id>
+```
+
+| Option | Required? | Description |
+|--------|-----------|-------------|
+| `--function` or `-f` | Yes | The AWS Lambda function name or ARN in which to add a layer. |
```
Comment (kolanos) [inline README.md]: Description needs to be updated here.

File: README.md [hunk 2]
```diff
@@ -132,6 +132,37 @@ newrelic-lambda functions list --filter installed
...
 
+
+### NewRelic Log Subscription
+
+#### Install Log Subscription
+
+```bash
+newrelic-lambda subscriptions install \
+    --function <name or arn> \
+    --nr-account-id <new relic account id>
```
Comment (kolanos) [inline README.md]: Don't believe this is necessary for the log subscription?

File: README.md [hunk 3]
```diff
@@ -132,6 +132,37 @@ newrelic-lambda functions list --filter installed
...
+
+#### Uninstall Log Subscription
+
+```bash
+newrelic-lambda subscriptions uninstall --function <name or arn>
+```
+
+| Option | Required? | Description |
+|--------|-----------|-------------|
+| `--function` or `-f` | Yes | The AWS Lambda function name or ARN in which to remove a layer. |
```
Comment (kolanos) [inline README.md]: Description update here as well.

File: newrelic_lambda_cli/cli/subscriptions.py [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+import click
+
+from .. import integrations, permissions
+from .cliutils import done
+from .decorators import add_options, AWS_OPTIONS
+
+
+@click.group(name="subscriptions")
+def subscriptions_group():
+    """Manage New Relic AWS Lambda Layers"""
```
Comment (kolanos) [inline newrelic_lambda_cli/cli/subscriptions.py]: Update docstring. This is what is displayed next to the group for `newrelic-lambda --help`.

File: newrelic_lambda_cli/cli/subscriptions.py [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+    "--function",
+    "-f",
+    help="AWS Lambda function name or ARN",
+    metavar="<arn>",
+    required=True,
+    type=click.STRING,
+)
+@click.pass_context
+def install(ctx, nr_account_id, aws_profile, aws_region, function):
+    """Install New Relic AWS Lambda Log Subsciption"""
```
Comment (kolanos) [inline newrelic_lambda_cli/cli/subscriptions.py]: Typo in this docstring.

File: newrelic_lambda_cli/cli/subscriptions.py [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+
+
+def register(group):
+    group.add_command(subscriptions_group)
+    subscriptions_group.add_command(install)
+    subscriptions_group.add_command(uninstall)
+
+
+@click.command(name="install")
+@click.option(
```
Comment (kolanos) [inline newrelic_lambda_cli/cli/subscriptions.py]: This option isn't necessary.

File: newrelic_lambda_cli/cli/subscriptions.py [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+    "--function",
+    "-f",
+    help="AWS Lambda function name or ARN",
+    metavar="<arn>",
+    required=True,
+    type=click.STRING,
+)
+@click.pass_context
+def install(ctx, nr_account_id, aws_profile, aws_region, function):
```
Comment (kolanos) [inline newrelic_lambda_cli/cli/subscriptions.py]: Be sure to remove `nr_account_id` here. Also unused.

File: newrelic_lambda_cli/cli/subscriptions.py [hunk 8]
```diff
@@ -0,0 +1,68 @@
+import json
```
Comment (kolanos) [inline newrelic_lambda_cli/cli/subscriptions.py]: Unused import.

**Pull request merged**